google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITForeignKeyDeleteCascadeTest.java

```diff
@@ -0,0 +1,540 @@
+      + "'";
+
+  private static final String QUERY_REFERENTIAL_CONSTRAINTS_V2 = "SELECT DELETE_RULE\n"
+      + "FROM INFORMATION_SCHEMA.REFERENTIAL_CONSTRAINTS\n"
+      + "WHERE CONSTRAINT_NAME =" + "\"" + CONCERTS_SINGER_FOREIGN_KEY_CONSTRAINT_NAME_V2
+      + "\"";
+  private static final String POSTGRES_QUERY_REFERENTIAL_CONSTRAINTS_V2 = "SELECT DELETE_RULE\n"
+      + "FROM INFORMATION_SCHEMA.REFERENTIAL_CONSTRAINTS\n"
+      + "WHERE CONSTRAINT_NAME =" + "'" + CONCERTS_SINGER_FOREIGN_KEY_CONSTRAINT_NAME_V2
+      + "'";
+
+  private static Database GOOGLE_STANDARD_SQL_DATABASE;
+  private static Database POST_GRE_SQL_DATABASE;
+  private static List<Database> dbs = new ArrayList<>();
+
+
+  @Parameterized.Parameter(0)
+  public DialectTestParameter dialect;
+
+  @BeforeClass
+  public static void setUpDatabase() {
+    GOOGLE_STANDARD_SQL_DATABASE =
+        env.getTestHelper()
+            .createTestDatabase(ImmutableList.of(
+                CREATE_TABLE_SINGER, CREATE_TABLE_CONCERT_WITH_FOREIGN_KEY));
+    if (!isUsingEmulator()) {
+      POST_GRE_SQL_DATABASE =
+          env.getTestHelper()
+              .createTestDatabase(
+                  Dialect.POSTGRESQL,
+                  ImmutableList.of(POSTGRES_CREATE_TABLE_SINGER,
+                      POSTGRES_CREATE_TABLE_CONCERT_WITH_FOREIGN_KEY));
+    }
+    dbs.add(GOOGLE_STANDARD_SQL_DATABASE);
+    dbs.add(POST_GRE_SQL_DATABASE);
+  }
+
+  @AfterClass
+  public static void tearDown() {
+    for (Database db : dbs) {
+      db.drop();
+    }
+    dbs.clear();
+  }
+
+  @Test
+  public void testForeignKeyDeleteCascadeConstraints_withCreateDDLStatements() {
+    final DatabaseClient databaseClient = getCreatedDatabaseClient();
+    final String referentialConstraintQuery = getReferentialConstraintsQueryStatement();
+    try (final ResultSet rs =
+        databaseClient
+            .singleUse()
+            .executeQuery(Statement.of(referentialConstraintQuery))) {
+      while (rs.next()) {
+        assertThat(rs.getString(DELETE_RULE_COLUMN_NAME)).isEqualTo(DELETE_RULE_CASCADE);
+      }
+    }
+  }
+
+  @Test
+  public void testForeignKeyDeleteCascadeConstraints_withAlterDDLStatements() throws Exception {
+    // Creating new tables within this test to ensure we don't pollute tables used by other tests in this class.
+    final List<String> createStatements = getCreateAndAlterTableStatementsWithForeignKey();
+    final Database createdDatabase = createDatabase(createStatements);
+    dbs.add(createdDatabase);
+
+    final DatabaseClient databaseClient = getCreatedDatabaseClient();
+
+    final String referentialConstraintQuery = getReferentialConstraintsQueryStatement();
+    try (final ResultSet rs =
+        databaseClient
+            .singleUse()
+            .executeQuery(
+                Statement.of(referentialConstraintQuery))) {
+      while (rs.next()) {
+        assertThat(rs.getString(DELETE_RULE_COLUMN_NAME)).isEqualTo(DELETE_RULE_CASCADE);
+      }
+    }
```

Comment:
For all the tests apart from `testForeignKeyDeleteCascadeConstraints_withAlterDDLStatements` the DDL statements are within the `setUpDatabase` method which is executed once per class. Hence, for majority of the tests we won't be able to make the SQL statement inline with the test method.

For `testForeignKeyDeleteCascadeConstraints_withAlterDDLStatements` because we had to execute a different set of DDL statements those were put in private methods `getCreateAndAlterTableStatementsWithForeignKey` which is just a replacement of `setUpDatabase` method.

I can still think to eliminate the private method `getCreateAndAlterTableStatementsWithForeignKey` and make statements inline with the test method but that will just eliminate the concern with 1 test method. For the remaining tests we will still need to scroll to `setUpDatabase` method in order to understand the DDL statement. I'll also eliminate the `getCreateAndAlterTableStatementsWithForeignKey` method and eliminate a few of the constants but the core readability concern of having to scroll different methods to understand test context will still remain.